### PR TITLE
Metadata update

### PR DESCRIPTION
**Pull request by oberonia78**
This PR updates the metadata fields. Previously, the metadata was extracted from the HDF5 file, but this version will read them from GeoTiff. 
Also, there are many changes in metadata fields including RFI etc.

### CODE REVIEW

File: src/dswx_sar/metadata.py [hunk 1]
```diff
@@ -266,24 +337,54 @@ def _get_general_dswx_metadata_dict(cfg, product_version=None):
...
     dswx_metadata_dict['PROCESSING_DATETIME'] = datetime.now().strftime("%Y-%m-%dT%H:%M:%SZ")
-    dswx_metadata_dict['POLARIZATION'] = cfg.groups.processing.polarizations
 
     return dswx_metadata_dict
 
 
-def gather_rtc_files(rtc_dirs):
+def gather_rtc_files(rtc_dirs, pol):
     """
     Given directories containing RTC files, gather all h5 files.
```
Comment (seongsujeong) [inline src/dswx_sar/metadata.py:346]:
```suggestion
    Given directories containing RTC files, gather all TIF files of the polarization `pol`.
    
    Parameters
    ------------
    rtc_dirs: str
        RTC output directory
    pol: str:
        polarization
    
    Returns:
    ---------
    tif_files: list
        List of RTC output TIF files in the polarization `pol`
```

Did not check the indentation in the suggestion above.

File: src/dswx_sar/metadata.py [hunk 2]
```diff
@@ -266,24 +337,54 @@ def _get_general_dswx_metadata_dict(cfg, product_version=None):
...
     """
-    h5_files = []
+    tif_files = []
     for rtc_input_dir in rtc_dirs:
-        rtc_h5_file = glob.glob(os.path.join(rtc_input_dir, '*.h5'))
-        if rtc_h5_file:
-            h5_files.append(rtc_h5_file[0])
-    return h5_files
+        rtc_tif_file = glob.glob(os.path.join(rtc_input_dir, f'*{pol}*.tif'))
```
Comment (seongsujeong) [inline src/dswx_sar/metadata.py:350]:
```suggestion
        rtc_tif_file = glob.glob(os.path.join(rtc_input_dir, f'*{pol.upper()}*.tif'))
```

Another suggestion: what do you think about using [glob.iglob](https://docs.python.org/3/library/glob.html#glob.iglob)?

File: src/dswx_sar/save_mgrs_tiles.py [hunk 3]
```diff
@@ -649,10 +650,27 @@ def run(cfg):
                 no_data=no_data_raster)
 
     # Get list of MGRS tiles overlapped with mosaic RTC image
+    mgrs_meta_dict = {}
+
     if database_bool:
-        mgrs_tile_list = get_intersecting_mgrs_tiles_list_from_db(
+        mgrs_tile_list, most_overlapped = get_intersecting_mgrs_tiles_list_from_db(
```
Comment (seongsujeong) [inline src/dswx_sar/save_mgrs_tiles.py:656]: Cannot post the comment directly to `get_intersecting_mgrs_tiles_list_from_db()`, but it seems `most_overlapped` is not mentioned in the docstring. Suggest revising the docstring.

**Review by seongsujeong — COMMENTED** [review]
My comments and suggestions are as below - S

**Comment by seongsujeong** [comment]
Another comment. Looks like CI is failing for some reason. Can you take care of the issue?

File: src/dswx_sar/metadata.py [hunk 4]
```diff
@@ -26,69 +29,96 @@
...
+            for rtc_field, dswx_field in dswx_meta_mapping.items():
+                rtc_meta_content = tags[rtc_field]
+
+                if rtc_field == 'INPUT_L1_SLC_GRANULES':
+                    rtc_meta_content = rtc_meta_content[2:-2]
+                # rtc_meta_content = ast.literal_eval(tags[rtc_field])
+                # rtc_meta_content = ast.literal_eval(tags[rtc_field])
+                # # if isinstance(rtc_meta_content, list):
+                # #     rtc_meta_content = str(rtc_meta_content)
+                # metadata_dict[rtc_field].append(rtc_meta_content)
```
[the remark this answers is not in the document]
Comment (oberonia78) [reply src/dswx_sar/metadata.py:73]: Good catch. Forgot to delete it

**Review by seongsujeong — APPROVED** [review]
LGTM

**Pull request merged**